### PR TITLE
Feature/create todo item

### PR DESCRIPTION
**Pull request by gagan-maheshwari**
## Description
Wrote the function to create a new Todo item

## Database schema changes
1. Created a new database: todo-app-db
2. Added a new table: todos

### Automated test cases added
1. should create a new todo item
2. should disallow blank todo item
3. should disallow any duplicate todo item

### Manual test cases run

*Pre-requisites*
1. Run the application using `npm run build` followed by `npm run start`
2. Open Postman and create new request with the following details:
> 1. Method Type: POST
> 2. URL: http://localhost:4200/todo/todos
> 3. In the Body section, select `raw` and `JSON`

**Test Case 1**
Verify the happy case.

1. Enter the data as:
`
{
 "title": "New Todo"
}
`
2. Hit `Send` button
3. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 201 Created
> A JSON containing the id and the title should come up
---
**Test Case 2**
Verify the duplicate input.

1. Execute test case 1.
2. Enter the data as:
`
{
 "title": "New Todo"
}
`
3. Hit `Send` button
4. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 400 Bad Request
> A JSON containing the failure message should come up as: `This todo item already exists. Please try a new one.`
---
**Test Case 3**
Verify for blank input.

1. Enter the data as:
`
{
 "title": ""
}
`
2. Hit `Send` button
3. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 400 Bad Request
> A JSON containing the failure message should come up as: `Please enter a valid todo item`
---

### CODE REVIEW

File: src/locales/en.json [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-    "INVALID_EMAIL": "Please specify valid email address",
-    "INVALID_FIRSTNAME": "Please specify first name.",
-    "INVALID_LASTNAME": "Please specify last name.",
-    "PASSWORD_MISSING": "Please provide a password.",
-    "DUPLICATE_EMAIL": "The email you provided is already taken. Please provide a different one."
-  },
-  "MESSAGES": {
-    "HEALTHCHECK": "OK"
-  }
+	"DEFAULT_ERRORS": {
```
Comment (jjalan) [inline src/locales/en.json:2]: 1. Use 2 instead of 4 spaces

File: test/mocks/app-context.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { Repositories } from '@storage';
-import { AppContext } from '@typings';
+import { Repositories } from '../../src/storage';
```
Comment (jjalan) [inline test/mocks/app-context.ts:1]: Why are we changing them to relative path - here and below?

File: test/spec/todo/todo.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const res = await chai
+      .request(expressApp)
+      .post("/todo/todos")
+      .send({
+        title: "ATC: Todo Item"
+      });
+
+    expect(res).to.have.status(201);
+    expect(res.body).to.have.property("id");
+    expect(res.body).to.have.property("title");
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts:34]: While writing tests, we should have 3 steps:

1. Pre-condition (i.e. no todo item exists in database with the said title). We should assert this.
2. Execute the text (which is what you are doing)
3. Post-condition - ensure that the data is in database.

For example, one may write the controller to return this response without storing in database and test would not capture that

**Review by jjalan — CHANGES_REQUESTED** [review]

File: src/locales/en.json (hunk 1 above)
Comment (gagan-maheshwari) [reply src/locales/en.json:2]: Shall be rectified with the next PR

File: test/mocks/app-context.ts (hunk 2 above)
Comment (gagan-maheshwari) [reply test/mocks/app-context.ts:1]: Somehow, it was not working with the @ path style. Hence had to resort to this style.

File: test/spec/todo/todo.spec.ts (hunk 3 above)
Comment (gagan-maheshwari) [reply test/spec/todo/todo.spec.ts:34]: Ok. Shall keep in mind.

**Pull request merged**